modules/dailyhuntBidAdapter.js

```diff
@@ -0,0 +1,193 @@
+        };
+      } else {
+        let ortbReq = {
+          id: bidderRequest.auctionId,
+          imp: [{
+            id: i.toString(),
+            banner: {
+              id: 'banner-' + bidderRequest.auctionId,
+              format: [
+                {
+                  'h': 250,
+                  'w': 300
+                },
+                {
+                  'h': 50,
+                  'w': 320
+                }
+              ]
+            },
+            bidfloor: 0,
+            ext: {
+              dailyhunt: {
+                ...params
+              }
+            }
+          }],
+          site: { id: i.toString(), page },
+          device: { userAgent },
+          user: {
+            id: params.clientId || '',
+          }
+        };
+        request = {
+          method: 'POST',
+          url: PROD_PREBID_ENDPOINT_URL,
+          data: JSON.stringify(ortbReq),
+          options: {
+            contentType: 'application/json',
+            withCredentials: true
+          },
+          bids: validBidRequests
+        };
+      }
+      serverRequests.push(request);
+    });
+    return serverRequests;
+  },
+
+  interpretResponse: function (serverResponse, request) {
+    let bidResponses = [];
+    if (!request.bids) {
+      let bid = serverResponse.body[0][0].ad;
+      if (bid.typeId != 2 && bid.typeId != 3) {
+        return bidResponses;
+      }
+      let impTrackers = [];
+      impTrackers.push(bid.beaconUrl);
+      impTrackers = (bid.beaconUrlAdditional && bid.beaconUrlAdditional.length !== 0) ? impTrackers.concat(bid.beaconUrlAdditional) : impTrackers;
+      let bidResponse = {
+        requestId: bid.pbRequestId,
+        cpm: bid.price,
+        creativeId: bid.bannerid,
+        currency: 'USD',
+        ttl: 360,
+        netRevenue: true,
+      };
+      bidResponse.mediaType = 'native'
+      bidResponse.native = {
+        title: bid.content.itemTitle.data,
+        body: bid.content.itemSubtitle1.data,
+        body2: bid.content.itemSubtitle1.data,
+        cta: bid.content.itemSubtitle2.data,
+        clickUrl: _encodeURIComponent(bid.action),
+        impressionTrackers: impTrackers,
+        clickTrackers: bid.landingUrlAdditional && bid.landingUrlAdditional.length !== 0 ? bid.landingUrlAdditional : [],
+        image: {
+          url: bid.content.iconLink,
+          height: bid.height,
+          width: bid.width
+        },
+        icon: {
+          url: bid.content.iconLink,
+          height: bid.height,
+          width: bid.width
+        }
+      }
+      bidResponses.push(bidResponse);
+      return bidResponses;
+    } else {
+      if (!serverResponse.body) {
+        return;
+      }
+      const { seatbid } = serverResponse.body;
+      let bids = request.bids
+      return bids.map((bid, id) => {
+        const _cbid = seatbid && seatbid[id] && seatbid[id].bid;
+        const bidResponse = _cbid && _cbid[0];
+        if (bidResponse) {
+          let obj = {
```

Comment:
Doesn't look like assignment to `obj` is needed. Can `return` be moved to this line?

Reply:
Changes Fixed.

Reply:
In making this change, you also added an `index` variable.  

You can have `id` as a parameter in the `reduce` function, similar to how it was before in `map`. It would be the 3rd param instead of the 2nd.

Reply:
Changes Fixed.